test/strategy/StargateV2Strategy.t.sol

```diff
@@ -0,0 +1,725 @@
+            amount *= 10 ** ERC20(assetData[i].asset).decimals();
+            deal(assetData[i].asset, VAULT, amount, true);
+            ERC20(assetData[i].asset).approve(address(strategy), amount);
+            strategy.deposit(assetData[i].asset, amount);
+        }
+    }
+
+    // Mock Utils:
+    function _mockInsufficientRwd(address asset) internal {
+        // Do a time travel & mine dummy blocks for accumulating some rewards
+        vm.warp(block.timestamp + 10 days);
+        vm.roll(block.number + 1000);
+
+        (, uint256[] memory pendingRewards) = strategy.checkPendingRewards(asset);
+        assert(pendingRewards[0] > 0);
+
+        // MOCK: Withdraw rewards from the farm.
+        changePrank(STARGATE_REWARDER);
+        ERC20(E_TOKEN).transfer(actors[0], ERC20(E_TOKEN).balanceOf(STARGATE_REWARDER));
+        changePrank(currentActor);
+    }
+
+    function _configAsset() internal {
+        assetData.push(
+            AssetData({
+                name: "USDT",
+                asset: 0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9,
+                pool: 0x8D66Ff1845b1baCC6E87D867CA4680d05A349cA8
+            })
+        );
+
+        assetData.push(
+            AssetData({
+                name: "USDC",
+                asset: 0xaf88d065e77c8cC2239327C5EDb3A432268e5831,
+                pool: 0x6Ea313859A5D9F6fF2a68f529e6361174bFD2225
+            })
+        );
+    }
+}
+
+contract Test_Initialization is StargateStrategyV2Test {
+    function test_ValidInitialization() public useKnownActor(USDS_OWNER) {
+        // Test state variables pre initialization
+        assertEq(impl.owner(), address(0));
+        assertEq(strategy.owner(), address(0));
+
+        // Initialize strategy
+        _initializeStrategy();
+
+        // Test state variables post initialization
+        assertEq(impl.owner(), address(0), "Implementation has a valid owner");
+        assertEq(strategy.owner(), USDS_OWNER, "Owner is not set correctly");
+        assertEq(strategy.vault(), VAULT, "Vault not correct");
+        assertEq(strategy.rewarder(), STARGATE_REWARDER, "Rewarder not correct");
+        assertEq(strategy.farm(), STARGATE_FARM, "Farm not correct");
+        assertEq(strategy.depositSlippage(), BASE_DEPOSIT_SLIPPAGE, "Deposit slippage not correct");
+        assertEq(strategy.withdrawSlippage(), BASE_WITHDRAW_SLIPPAGE, "Withdraw slippage not correct");
+    }
+
+    function test_InvalidInitialization() public useKnownActor(USDS_OWNER) {
+        vm.expectRevert(abi.encodeWithSelector(Helpers.InvalidAddress.selector));
+        strategy.initialize(address(0), address(0), STARGATE_FARM, BASE_DEPOSIT_SLIPPAGE, BASE_WITHDRAW_SLIPPAGE);
+    }
+
+    function test_UpdateVaultCore() public useKnownActor(USDS_OWNER) {
+        _initializeStrategy();
+
+        address newVault = address(1);
+        vm.expectEmit(address(strategy));
+        emit VaultUpdated(newVault);
+        strategy.updateVault(newVault);
```

Comment:
we should assert new vault. Since there could be a case where we emit the right data but set wrong data.

Reply:
Applies to all the rest of the cases too.